File: web/src/integration/titaniumdrm/TitaniumWidevineContentProtectionIntegrationFactory.ts
import {
    ContentProtectionIntegration,
    ContentProtectionIntegrationFactory
} from 'THEOplayer';
import { TitaniumWidevineContentProtectionIntegration } from './TitaniumWidevineContentProtectionIntegration';
import { TitaniumDrmIntegrationConfiguration } from './TitaniumDrmIntegrationConfiguration';

export class TitaniumWidevineContentProtectionIntegrationFactory implements ContentProtectionIntegrationFactory {
    build(configuration: TitaniumDrmIntegrationConfiguration): ContentProtectionIntegration {
        return new TitaniumWidevineContentProtectionIntegration(configuration);
    }
}

File: android/app/src/main/java/com/theoplayer/contentprotectionintegration/integration/titanium/TitaniumWidevineContentProtectionIntegrationFactory.java
package com.theoplayer.contentprotectionintegration.integration.titanium;

import com.theoplayer.android.api.contentprotection.ContentProtectionIntegration;
import com.theoplayer.android.api.contentprotection.ContentProtectionIntegrationFactory;
import com.theoplayer.android.api.source.drm.DRMConfiguration;

public class TitaniumWidevineContentProtectionIntegrationFactory implements ContentProtectionIntegrationFactory {
    @Override
    public ContentProtectionIntegration build(DRMConfiguration configuration) {
        return new TitaniumWidevineContentProtectionIntegration(configuration);
    }
}
